src/tokenizer.rs

```diff
@@ -53,6 +55,39 @@ impl<R: Reader> Tokenizer<R> {
     }
 }
 
+/// The states you can set the tokenizer to.
+#[derive(Debug)]
```

Comment:
I'm happy to expose the full enum, seems like that's generally required to build meaningful extensions on top of the tokenizer.

Reply:
I think it's good practice to keep public APIs minimal. Marking it as `#[non_exhaustive]` for now would let us expose the whole enum in the future if the need arises ... (which I currently doubt). But it's your call ... if you are certain that you want to expose the full enum I can do that.

Reply:
I'm happy to do that for now as long as the crate is pre-1.0. If I change my mind I'll just push another 0.x version. But I think the enum is stable because the spec is.

The crate is still very in-flux, I have no idea what a proper tree-builder on top of it could look like, but I think this step is necessary to make it as useful as possible to people right now.

Reply:
You don't have to decide that before 1.0. For now the state enum is marked as `#[non_exhaustive]` so you could expose the whole state enum after 1.0 without needing a breaking change.

Reply:
sounds good, thanks